### PR TITLE
WIP Changes to API and Errors for tutorial updates

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
This PR includes a number of API changes;
- Update main to include a msg when exiting, e.g. `Task.err (Exit -2 "this msg be printed to stderr")`
- Update errors for modules to a common pattern `[CmdError Err]` using an `Err` type. 
 
Note this PR depends on the following
- Merge https://github.com/roc-lang/roc/pull/6676

### CODE REVIEW

File: platform/Cmd.roc [hunk 1]
```diff
@@ -24,7 +25,14 @@ interface Cmd
...
 
 ## Errors from executing a command.
-Error : InternalCommand.CommandErr
+Err : InternalCommand.CommandErr
+
+outputErrToStr : (Output, Err) -> Str
+outputErrToStr = \err ->
+    when err is
+        (_, (ExitCode code)) -> "Child exited with non-zero code: $(Num.toStr code)"
+        (_, (KilledBySignal)) -> "Child was killed by signal"
```
Comment (joshuawarner32) [inline platform/Cmd.roc:34]: Heads up: `(KilledBySignal)` is not really a tuple (just a parenthesized expr), if that's what you had intended. Seems kinda misleading to parenthesize it.

Comment (lukewilliamboswell) [reply platform/Cmd.roc]: Thanks, updated

**Comment by rtfeldman** [comment]
https://github.com/roc-lang/roc/pull/6676 has now landed, so once the new nightly comes out we should be able to re-run these tests!

**Comment by Anton-4** [comment]
I'm not sure why this happens on ci_nix with latest Roc main:
```
 spawn ./examples/dir
Program exited early with error: (UnableToCreateFirstDir (DirError WriteZero))
```

**Comment by Anton-4** [comment]
The dir example was failing before but in a way that slipped past the tests.

**Comment by lukewilliamboswell** [comment]
Is it because we changed the directory name from "a" to "dirExampleA"? We haven't had to whitelist the ability to write inside nix or something? It's strange that Rust cant write any bytes to the file.

**Comment by Anton-4** [comment]
Don't merge this yet, I think the tee command I just added is messing with the exit code

**Comment by Anton-4** [comment]
The dir stuff is behaving strangely, if I add a dbg in the platform it works normally. It also has valgrind errors:
```
❯ rm -rf dirExampleE/ && valgrind examples/dir
==126467== Memcheck, a memory error detector
==126467== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==126467== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==126467== Command: examples/dir
==126467== 
==126467== Conditional jump or move depends on uninitialised value(s)
==126467==    at 0x1147E10: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467==    by 0x114A150: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467==    by 0x1FFEFEF28F: ???
==126467==    by 0x4A600B7: ???
==126467==    by 0x4034AAF: ??? (in /nix/store/ddwyrxif62r8n6xclvskjyy6szdhvj60-glibc-2.39-5/lib/ld-linux-x86-64.so.2)
==126467==    by 0x114B9D2: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467==    by 0x1FFEFEF447: ???
==126467==    by 0x1FFEFEF2BF: ???
==126467==    by 0x4: ???
==126467==    by 0x4A694CF: ???
==126467==    by 0x8: ???
==126467==    by 0x1145E32: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467== 
==126467== Conditional jump or move depends on uninitialised value(s)
==126467==    at 0x1142FCD: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467==    by 0x11444A0: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467==    by 0x1144359: ??? (in /home/username/gitrepos/basic-cli/examples/dir)
==126467== 
Program exited early with error: (UnableToCreateFirstDir (DirError WriteZero))
==126467== 
==126467== HEAP SUMMARY:
==126467==     in use at exit: 0 bytes in 0 blocks
==126467==   total heap usage: 16 allocs, 16 frees, 2,415 bytes allocated
==126467== 
==126467== All heap blocks were freed -- no leaks are possible
==126467== 
==126467== Use --track-origins=yes to see where uninitialised values come from
==126467== For lists of detected and suppressed errors, rerun with: -s
==126467== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
```
Does the glue need to be updated?

I'm also fine with cutting the dir API from basic-cli if need be.

I also ran all the tests without the dir stuff and everything works, so that's the last thing here.
See you in the morning :wave:

**Review by rtfeldman — APPROVED** [review]

**Pull request merged**